### ISSUE TITLE
SSD Not Recognized

### ISSUE THREAD

**Issue by jakesthetester** [openhardwaremonitor/openhardwaremonitor#1263]
My main system drive SSD (Samsung 970 EVO Plus 250GB) is not being recognized by any of the last 3 iterations of this software.  All storage drives are still configured exactly as they arrived when they were brand new.  I have another computer on the network that has no problem exposing its SSD (also a Samsung) to OHM.  Will there be a patch or release that can fix this, or am I doing something wrong?  I have included a copy of my .txt file for reference...

[OpenHardwareMonitor.Report.txt](https://github.com/openhardwaremonitor/openhardwaremonitor/files/4562126/OpenHardwareMonitor.Report.txt)

**Comment by Claire83310** [comment]
Same problem here with my SSD Samsung 970 EVO Plus 500GB
Report & Smart here

[OpenHardwareMonitor.Report.txt](https://github.com/openhardwaremonitor/openhardwaremonitor/files/4580457/OpenHardwareMonitor.Report.txt)
![4](https://user-images.githubusercontent.com/64842509/81062548-72e55880-8ed6-11ea-83e6-d68e2b0576d9.JPG)
![1](https://user-images.githubusercontent.com/64842509/81062551-737def00-8ed6-11ea-8b61-2047d4a33a1a.JPG)
![2](https://user-images.githubusercontent.com/64842509/81062555-74168580-8ed6-11ea-81da-9f1520a1f83a.JPG)
![3](https://user-images.githubusercontent.com/64842509/81062557-74af1c00-8ed6-11ea-8c29-e5e2a13d1fa9.JPG)

**Comment by ahmadSum1** [comment]
Same here with my m.2 nvme SSD, The other SATA 2.5" SSD was recognized properly.
SSD not properly recognized. The "Generic Hard Disk (/hdd/2)" is actually a Gigabyte GP-GSM2NE3128GNTD 128GB,
Firmware: EDFM00.2
[OpenHardwareMonitor.Report_23May2020.txt](https://github.com/openhardwaremonitor/openhardwaremonitor/files/4669646/OpenHardwareMonitor.Report_23May2020.txt)

**Comment by flexicart** [comment]
Hello, same issue here, glad I'm not alone! :)

[OpenHardwareMonitor.Report.txt](https://github.com/openhardwaremonitor/openhardwaremonitor/files/4814851/OpenHardwareMonitor.Report.txt)
![hdd not reading](https://user-images.githubusercontent.com/5055912/85320081-d5180e00-b4c2-11ea-8160-22968f1f4d31.png)

**Comment by Sopor** [comment]
I have the same issue here with my two M.2 drives. OHM will only show Generic hard drive

Speecy seems to detect some of the info but not all as you can see below
```
476GB Samsung SSD 970 PRO 512GB (Unknown (SSD))
465GB NVMe Samsung SSD 970 SCSI Disk Device (Unknown (SSD))
```

**Comment by jpstotz** [comment]
It seems like OpenHardwareMonitor does not yet support NVMe SSDs (Windows driver).

NVMe SSDs require different code that regular SATA drives for retrieving detailed info. Example code how to query e.g. the NVMe temperature can be found here: https://stackoverflow.com/a/55241041/150978

**Comment by farp96** [comment]
AW that's why mine isn't read. OH well back to HW monitor.

**Comment by MrCanan** [comment]
Hi all,
Same for me, OpenHardwareMonitor doesn't support NVMe SSDs, so I got two lines "Generic Hard Disk".
Any news about his support?
Best regards.

**Comment by jpstotz** [comment]
There is pull request #984 (opened in 2017) which adds support for NVMe drives that are not using the standard Windows NVMe driver. 

It it seems like there were problems because this PR was never included into OpenHardwareMonitor.

**Comment by donpin1** [comment]
Agree, please add NVMe drives and their temperatures. It would be great if you also had a tab that monitored SMART attributes. Nevertheless, this is easily the best program for me that I have come across. Well done.
![Capture](https://user-images.githubusercontent.com/79115442/108460770-37d5d980-72de-11eb-94b2-e27843567ac4.JPG)
